controllers/dbaas.redhat.com/dbaasprovider_reconciler.go

```diff
@@ -0,0 +1,286 @@
+	MEDIATYPE              = "image/svg+xml"
+	KEYFIELDNAME           = "publicApiKey"
+	KEYFIELDDISPLAYNAME    = "Public API Key"
+	SECRETFIELDNAME        = "privateApiSecret"
+	SECRETFIELDDISPLAYNAME = "Private API Secret"
+	RELATEDTOLABELNAME     = "related-to"
+	RELATEDTOLABELVALUE    = "dbaas-operator"
+	TYPELABELNAME          = "type"
+	TYPELABELVALUE         = "dbaas-provider-registration"
+	DBAASPROVIDERKIND      = "DBaaSProvider"
+)
+
+var labels = map[string]string{RELATEDTOLABELNAME: RELATEDTOLABELVALUE, TYPELABELNAME: TYPELABELVALUE}
+
+type DBaaSProviderReconciler struct {
+	client.Client
+	*runtime.Scheme
+	Log                      logr.Logger
+	Clientset                *kubernetes.Clientset
+	operatorNameVersion      string
+	operatorInstallNamespace string
+}
+
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;create;update;delete;watch
+// +kubebuilder:rbac:groups=dbaas.redhat.com,resources=dbaasproviders,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=dbaas.redhat.com,resources=dbaasproviders/status,verbs=get;update;patch
+
+func (r *DBaaSProviderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	log := r.Log.WithValues("during", "DBaaSProvider Reconciler")
+
+	// due to predicate filtering, we'll only reconcile this operator's own deployment when it's seen the first time
+	// meaning we have a reconcile entry-point on operator start-up, so now we can create a cluster-scoped resource
+	// owned by the operator's ClusterRole to ensure cleanup on uninstall
+
+	dep := &v1.Deployment{}
+	if err := r.Get(ctx, req.NamespacedName, dep); err != nil {
+		if errors.IsNotFound(err) {
+			// CR deleted since request queued, child objects getting GC'd, no requeue
+			log.Info("deployment not found, deleted, no requeue")
+			return ctrl.Result{}, nil
+		}
+		// error fetching deployment, requeue and try again
+		log.Error(err, "error fetching Deployment CR")
+		return ctrl.Result{}, err
+	}
+
+	isCrdInstalled, err := r.checkCrdInstalled(dbaasoperator.GroupVersion.String(), DBAASPROVIDERKIND)
+	if err != nil {
+		log.Error(err, "error discovering GVK")
+		return ctrl.Result{}, err
+	}
+	if !isCrdInstalled {
+		log.Info("CRD not found, requeueing with rate limiter")
+		// returning with 'Requeue: true' will invoke our custom rate limiter seen in SetupWithManager below
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	instance := &dbaasoperator.DBaaSProvider{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: NAME,
+		},
+	}
+	if err := r.Get(ctx, client.ObjectKeyFromObject(instance), instance); err != nil {
+		if errors.IsNotFound(err) {
+			// CR deleted since request queued, child objects getting GC'd, no requeue
```

Comment:
Delete this comment

Reply:
Ehh, that's being a bit heavy-handed in review :) There's nothing wrong with having this comment to clarify the state